Marlin/Conditionals.h

```diff
@@ -447,6 +447,7 @@
   #define HAS_Z2_MIN (PIN_EXISTS(Z2_MIN))
   #define HAS_Z2_MAX (PIN_EXISTS(Z2_MAX))
   #define HAS_Z_PROBE (PIN_EXISTS(Z_MIN_PROBE))
+  #define HAS_SOLENOID_0 (PIN_EXISTS(SOL0))
```

Comment:
It was assumed that at least one solenoid (Solenoid 0) existed, because you had to enable `EXT_SOLENOID` in order to define M380 and M381. I don't think the solenoid was previously considered an extruder.

`EXT_SOLENOID` was not defined anywhere, and in fact, the case statement didn't even deal with M380/381. The functionality has now been coupled with `PNEUMATICS`